controllers/podplacement/scheduling_gate_mutating_webhook.go

```diff
@@ -66,13 +66,19 @@ func (a *PodSchedulingGateMutatingWebHook) Handle(ctx context.Context, req admis
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	// ignore the openshift-* namespace as those are infra components, and ignore the namespace where the operand is running too
-	if utils.Namespace() == pod.Namespace || strings.HasPrefix(pod.Namespace, "openshift-") ||
-		strings.HasPrefix(pod.Namespace, "hypershift-") || strings.HasPrefix(pod.Namespace, "kube-") ||
-		pod.Spec.NodeName != "" {
+	// ignore the kube-* and hypershift-* namespace as those are infra components, and ignore the namespace where the operand is running too
+	if utils.Namespace() == pod.Namespace || strings.HasPrefix(pod.Namespace, "hypershift-") ||
+		strings.HasPrefix(pod.Namespace, "kube-") || pod.Spec.NodeName != "" {
```

Comment:
```suggestion
		strings.HasPrefix(pod.Namespace, "kube-") || pod.Spec.NodeName != "" ||
		pod.Spec.NodeSelector != nil && utils.HasControlPlaneNodeSelector(pod.Spec.NodeSelector) {
```
what about merging 75-81 here?

Reply:
yup can do - just didn't want to make that conditional longer :)